mdw-services/src/com/centurylink/mdw/service/data/task/TaskDataAccess.java

```diff
@@ -726,10 +729,10 @@ private String buildTaskInstanceWhere(Query query) throws DataAccessException {
                     if (!Workgroup.COMMON_GROUP.equals(workgroups[i])) {
                         Workgroup group = UserGroupCache.getWorkgroup(workgroups[i]);
                         if (group == null)
-                            throw new CachingException("Cannot find workgroup: " + workgroups[i]);
+                            throw new ServiceException("unable to find the workgroup: " + workgroups[i]);

```

Comment:
Please throw ServiceException with NOT_FOUND parameter so that 404 will be returned to caller:
```java
throw new ServiceException(ServiceException.NOT_FOUND, "Workgroup not found: " + workgroups[i]);
```
Thanks, Don

Reply:
thanks for pointing this out. implemented the above.